src/MainWindow.vala

```diff
@@ -236,13 +233,29 @@ namespace PantheonTerminal {
             destroy.connect (on_destroy);
 
             restorable_terminals = new HashTable<string, TerminalWidget> (str_hash, str_equal);
+
+            if (recreate_tabs) {
+                open_tabs ();
+            }
         }
 
         public void add_tab_with_command (string command) {
             new_tab ("", command);
         }
 
-        public void add_tab_with_working_directory (string location) {
+        public void add_tab_with_working_directory (string location, bool allow_duplicate = false) {
```

Comment:
Why do we need this `allow_duplicate` here? I do not see any calls to this function overriding this argument with `true`.

Reply:
I cannot remember tbh.  Maybe I was just unsure whether there might be a future need to add a duplicate tab from the commandline e.g. with a new flag.  I guess we can remove this parameter for now.